spec/unit/puppet-lint/plugins/top_scope_facts/top_scope_facts_spec.rb

```diff
@@ -102,6 +102,18 @@
       end
     end
 
+    context 'top scope structured fact not present on allowlist' do
+      let(:code) { "$::my_structured_fact['foo']['test']" }
```

Comment:
Will this now correct top scope variables to facts and break the code?

Reply:
It shouldn't, the original logic for this plugin should still apply. This particular fix will only trigger if a top scope structured fact such like `$::somefact['foo']` is detected

Reply:
does that make sense @bastelfreak? I'm happy to take on any concerns you have that this might cause further issues :)

Reply:
I think your change is fine.

But: The docs are a bit confusing:
> It assumes all top scope variables are facts. If you have top scope variables that aren't facts you should configure the [linter to ignore them](https://github.com/mmckinst/puppet-lint-top_scope_facts-check#configuring-the-check).

Okay. but why are there facts in the builtin allowlist? https://github.com/puppetlabs/puppet-lint/blob/e4863b0fda02c0bdc8f02448a41d9bc9852bb4f4/lib/puppet-lint/plugins/top_scope_facts/top_scope_facts.rb#L20

Does it allow legacy facts as topscope notation, e.g. `$::architecture`? or am I reading this wrong?

Reply:
@bastelfreak that allowlist was implemented as part of https://github.com/puppetlabs/puppet-lint/commit/3d7e9394aa91b9d0778e66ab0e1596b946922fff which fixed a bug were legacy facts with top scope notation were getting corrected twice.
I think the idea here is the `allow` them in this plugin, so that the legacy fact plugin can correct them appropriately. :)

Reply:
Ahhhh okay. makes sense now!